keras/ops/linalg.py

```diff
@@ -0,0 +1,596 @@
+from keras import backend
+from keras.api_export import keras_export
+from keras.backend import KerasTensor
+from keras.backend import any_symbolic_tensors
+from keras.ops.operation import Operation
+from keras.ops.operation_utils import reduce_shape
+
+
+class LinalgError(ValueError):
+    """Generic exception raised by linalg operations.
+
+    Raised when a linear algebra-related condition prevents the correct
+    execution of the operation.
+    """
+
+
+class Cholesky(Operation):
+    def __init__(self):
+        super().__init__()
+
+    def call(self, x):
+        return _cholesky(x)
+
+    def compute_output_spec(self, x):
+        _assert_2d(x)
+        _assert_square(x)
+        return KerasTensor(x.shape, x.dtype)
+
+
+@keras_export("keras.ops.linalg.cholesky")
```

Comment:
All these ops should be exported as both `keras.ops.cholesky` (first) and `keras.ops.linalg.cholesky` (like we do for e.g. `ops.nn`).

In general we don't do sub-namespaces. The `image` namespace is the only one, and it was only necessary because of naming ambiguity for `resize`.

Reply:
Done